mlir/utils/performance/perfRunner.py

```diff
@@ -1140,63 +1165,75 @@ def benchmarkFusionKernels(test_dir, paths: Paths, arch, numCU, tuningDb: MaybeT
+            partialCode, _ = runPipeline(commands)
+            mlirfile.seek(0)
+            mlirfile.write(bytes(partialCode))
+            mlirfile.seek(0)
+            tuningKeyCommand = [paths.mlir_paths.rocmlir_gen_path, '--emit-tuning-key', '-']
+            output, _ = runPipeline([tuningKeyCommand], initial_stdin=mlirfile)
+            result = output.decode('utf-8').strip().split('\t')
+            testVector = result[2]
+
+            print("Profiling:", filename)
+            # Sanity check
+            if not testVector:
+                print("\tCannot find a test vector")
+                continue
+            if not futName:
+                print("\tCannot find rocmlir-gen with -fut")
+                continue
 
-        commandLine = testVector.split(sep=' ')
-        if commandLine[0].startswith('conv'):
-            op = 'conv'
-            config = ConvConfiguration.fromCommandLine(commandLine, arch, numCU)
-        else:
-            op = 'gemm'
-            config = GemmConfiguration.fromCommandLine(commandLine, arch, numCU)
-
-        # Find the best perf_config
-        bestPerf =""
-        if tuningDb:
-            configStr = config.toCommandLine()
-            if (arch, configStr) in tuningDb:
-                bestPerf = tuningDb[arch, configStr]
-                config.setPerfConfig(bestPerf)
-            else: # Tuning DB present but doesn't contain config, add a NaN entry
-                if not testVector in perfResults:
-                    oneEntry = config.tableEntry(np.nan)
-                    oneEntry['MLIR TFlops'] = np.nan
-                    oneEntry['Fusion/MLIR'] = np.nan
-                    oneEntry['FileName'] = filename
-                    perfResults[testVector] = oneEntry
+            commandLine = testVector.split(sep=' ')
+            if commandLine[0].startswith('conv'):
+                op = 'conv'
+                config = ConvConfiguration.fromCommandLine(commandLine, arch, numCU)
+            else:
+                op = 'gemm'
+                config = GemmConfiguration.fromCommandLine(commandLine, arch, numCU)
+
+            # Find the best perf_config
+            bestPerf =""
+            if tuningDb:
+                configStr = config.toCommandLine()
+                if (arch, configStr) in tuningDb:
+                    bestPerf = tuningDb[arch, configStr]
+                    config.setPerfConfig(bestPerf)
+                else: # Tuning DB present but doesn't contain config, add a NaN entry
+                    if not testVector in perfResults:
+                        oneEntry = config.tableEntry(np.nan)
+                        oneEntry['MLIR TFlops'] = np.nan
+                        oneEntry['Fusion/MLIR'] = np.nan
+                        oneEntry['FileName'] = filename
+                        perfResults[testVector] = oneEntry
+                    continue
+
+            # Run fusion test
```

Comment:
Run the kernel, reusing the mlir in the temp file, and record its time.  We anticipate duplicates -- eg, in the bert tests there are 24 .mlir file but only eight unique kernels -- and just take the best-performing.  Given natural variation, times will be close but the winning file can be different every time.